File: create_test_user.js
#!/usr/bin/env node
/**
 * Create a test user for upload testing
 */

import axios from 'axios';
import { v4 as uuidv4 } from 'uuid';

const API_BASE_URL = 'http://localhost:8000';

async function createTestUser() {
    console.log('🚀 Creating test user for upload testing');
    console.log('========================================');
    
    // Test user credentials (matching test files)
    let testEmail = 'test@example.com';
    let testPassword = 'SecureP@ssw0rd123!';
    
    // Alternative test credentials to try
    const altTestEmail = 'testuser@example.com';
    const altTestPassword = 'SecureP@ssw0rd123!';
    
    console.log(`📧 Test email: ${testEmail}`);
    console.log(`🔑 Test password: ${testPassword}`);
    
    try {
        // Try login first
        console.log('\n🔸 Step 1: Trying login with existing test user...');
        const loginData = {
            email: testEmail,
            password: testPassword
        };
        
        try {
            const loginResponse = await axios.post(
                `${API_BASE_URL}/api/auth/login`,
                loginData,
                {
                    headers: { 'Content-Type': 'application/json' },
                    timeout: 30000
                }
            );
            
            if (loginResponse.status === 200) {
                console.log('✅ Login successful with existing user!');
                const loginResult = loginResponse.data;
                console.log(`   Access Token: ${loginResult.tokens.access_token.substring(0, 20)}...`);
                
                const credentials = {
                    email: testEmail,
                    password: testPassword,
                    access_token: loginResult.tokens.access_token,
                    user_id: loginResult.user.id
                };
                
                console.log('\n🎉 Test user login successful!');
                console.log('Use these credentials for upload testing:');
                console.log(`Email: ${testEmail}`);
                console.log(`Password: ${testPassword}`);
                console.log(`Token: ${loginResult.tokens.access_token.substring(0, 30)}...`);
                
                return credentials;
            }
        } catch (loginError) {
            console.log('❌ Login failed:', loginError.response?.status, loginError.response?.data);
            
            // Try alternative credentials
            console.log('\n🔸 Trying alternative credentials...');
            try {
                const altLoginResponse = await axios.post(`${API_BASE_URL}/api/auth/login`, {
                    email: altTestEmail,
                    password: altTestPassword
                });
                
                if (altLoginResponse.data && altLoginResponse.data.access_token) {
                    console.log('✅ Alternative login successful!');
                    console.log('📝 Credentials saved to test_credentials.json');
                    
                    // Save credentials for upload test
                    const credentials = {
                        email: altTestEmail,
                        password: altTestPassword,
                        access_token: altLoginResponse.data.access_token
                    };
                    
                    await fs.writeFile('test_credentials.json', JSON.stringify(credentials, null, 2));
                    return credentials;
                }
            } catch (altLoginError) {
                console.log('❌ Alternative login also failed:', altLoginError.response?.status, altLoginError.response?.data);
            }
            
            console.log('Will try registration...');
        }
        
        // If login failed, try registration with unique email
        testEmail = `test_${uuidv4().slice(0, 8)}@example.com`;
        console.log(`\n🔸 Step 2: Registering new test user: ${testEmail}`);
        const registrationData = {
            email: testEmail,
            password: testPassword,
            full_name: 'Upload Test User'
        };
        
        const registerResponse = await axios.post(
            `${API_BASE_URL}/api/auth/register`,
            registrationData,
            {
                headers: { 'Content-Type': 'application/json' },
                timeout: 30000
            }
        );
        
        if (registerResponse.status === 201) {
            console.log('✅ Registration successful!');
            const result = registerResponse.data;
            console.log(`   User ID: ${result.user.id}`);
            console.log(`   Email: ${result.user.email}`);
            console.log(`   Access Token: ${result.tokens.access_token.substring(0, 20)}...`);
            
            // Test login immediately
            console.log('\n🔸 Step 2: Testing login...');
            const loginData = {
                email: testEmail,
                password: testPassword
            };
            
            const loginResponse = await axios.post(
                `${API_BASE_URL}/api/auth/login`,
                loginData,
                {
                    headers: { 'Content-Type': 'application/json' },
                    timeout: 30000
                }
            );
            
            if (loginResponse.status === 200) {
                console.log('✅ Login successful!');
                const loginResult = loginResponse.data;
                console.log(`   Access Token: ${loginResult.tokens.access_token.substring(0, 20)}...`);
                
                // Save credentials for upload testing
                console.log('\n💾 Saving test credentials...');
                const credentials = {
                    email: testEmail,
                    password: testPassword,
                    access_token: loginResult.tokens.access_token,
                    user_id: loginResult.user.id
                };
                
                console.log('\n🎉 Test user created successfully!');
                console.log('Use these credentials for upload testing:');
                console.log(`Email: ${testEmail}`);
                console.log(`Password: ${testPassword}`);
                console.log(`Token: ${loginResult.tokens.access_token.substring(0, 30)}...`);
                
                return credentials;
            } else {
                console.log(`❌ Login failed: ${loginResponse.status}`);
                console.log(loginResponse.data);
            }
        } else {
            console.log(`❌ Registration failed: ${registerResponse.status}`);
            console.log(registerResponse.data);
        }
    } catch (error) {
        if (error.code === 'ECONNREFUSED') {
            console.log('❌ Connection refused - is the backend server running on port 8000?');
        } else if (error.response) {
            console.log(`❌ Request failed: ${error.response.status}`);
            console.log('Response data:', error.response.data);
        } else {
            console.log(`❌ Error: ${error.message}`);
        }
    }
}

// Debug output
console.log('Script loaded, checking execution context...');
console.log('process.argv[1]:', process.argv[1]);
console.log('import.meta.url:', import.meta.url);

// Run if this file is executed directly
const scriptPath = new URL(import.meta.url).pathname;
const isMainModule = process.argv[1].endsWith('create_test_user.js');
console.log('isMainModule:', isMainModule);

if (isMainModule) {
    console.log('Running createTestUser...');
    createTestUser();
} else {
    console.log('Script imported as module, not executing createTestUser');
}

export { createTestUser };

File: create_test_user.py
#!/usr/bin/env python3
"""
Create a test user for video upload testing
"""

import requests
import json
from datetime import datetime

# Supabase configuration
SUPABASE_URL = "https://stzdywhbdovjojtqxmsd.supabase.co"
SUPABASE_SERVICE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InN0emR5d2hiZG92am9qdHF4bXNkIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1NzU0MDU2OCwiZXhwIjoyMDczMTE2NTY4fQ.Xg6iDR-YMjID73OzQfJ362B-xYBadJeb6Qh2NxVtAmU"

def create_test_user():
    """Create a test user account"""
    print("🚀 Creating test user for video upload testing")
    
    # User data
    user_data = {
        "email": "test.upload@example.com",
        "password": "TestUpload123!",
        "email_confirm": True,
        "user_metadata": {
            "full_name": "Upload Test User",
            "role": "user"
        }
    }
    
    headers = {
        "Authorization": f"Bearer {SUPABASE_SERVICE_KEY}",
        "Content-Type": "application/json",
        "apikey": SUPABASE_SERVICE_KEY
    }
    
    try:
        # Create user via Supabase Auth Admin API
        response = requests.post(
            f"{SUPABASE_URL}/auth/v1/admin/users",
            headers=headers,
            json=user_data,
            timeout=10
        )
        
        if response.status_code == 200:
            user = response.json()
            print(f"✅ Test user created successfully!")
            print(f"   User ID: {user['id']}")
            print(f"   Email: {user['email']}")
            
            # Now sign in to get access token
            print("\n🔑 Signing in to get access token...")
            signin_data = {
                "email": user_data["email"],
                "password": user_data["password"]
            }
            
            signin_response = requests.post(
                f"{SUPABASE_URL}/auth/v1/token?grant_type=password",
                headers={
                    "Content-Type": "application/json",
                    "apikey": SUPABASE_SERVICE_KEY
                },
                json=signin_data,
                timeout=10
            )
            
            if signin_response.status_code == 200:
                tokens = signin_response.json()
                access_token = tokens['access_token']
                print(f"✅ Sign in successful!")
                print(f"   Access Token: {access_token[:50]}...")
                
                # Save credentials for testing
                test_credentials = {
                    "user_id": user['id'],
                    "email": user['email'],
                    "access_token": access_token,
                    "created_at": datetime.now().isoformat()
                }
                
                with open("test_user_credentials.json", "w") as f:
                    json.dump(test_credentials, f, indent=2)
                
                print("\n📝 Credentials saved to test_user_credentials.json")
                print("\nUse these credentials for upload testing:")
                print(f"   Authorization: Bearer {access_token}")
                
                return access_token
            else:
                print(f"❌ Sign in failed: {signin_response.status_code}")
                print(f"   Response: {signin_response.text}")
                return None
                
        else:
            print(f"❌ User creation failed: {response.status_code}")
            print(f"   Response: {response.text}")
            return None
            
    except Exception as e:
        print(f"❌ Error creating test user: {e}")
        return None

if __name__ == "__main__":
    access_token = create_test_user()
    if access_token:
        print("\n🎉 Test user setup completed successfully!")
        print("You can now use the access token for video upload testing.")
    else:
        print("\n❌ Test user setup failed.")